### PR TITLE
Feature/added n genius services

### PR DESCRIPTION
**Pull request by DFSko**

### CODE REVIEW

File: Api/Models/Payments/NGenius/BillingAddress.cs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+namespace HappyTravel.Edo.Api.Models.Payments.NGenius
+{
+    public struct BillingAddress
+    {
```
Comment (mingazhev) [inline Api/Models/Payments/NGenius/BillingAddress.cs:4]: NGeniusBillingAddress?

File: Api/Services/Payments/NGenius/NGeniusClient.cs [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+            var data = JsonSerializer.Deserialize<NGeniusAuthResponse>(await response.Content.ReadAsStringAsync());
+            _cache.Set(key, data.AccessToken, TimeSpan.FromSeconds(data.ExpiresIn).Subtract(TimeSpan.FromMinutes(1)));
+            token = data.AccessToken;
+
+            return token;
+        }
+
+
+        public async Task<Result<NGeniusPaymentResponse>> Authorize(OrderRequest order)
```
Comment (mingazhev) [inline Api/Services/Payments/NGenius/NGeniusClient.cs:51]: public methods must be upper than private

File: Api/Services/Payments/NGenius/NGeniusClient.cs [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+        {
+            var endpoint = $"transactions/outlets/{_options.OutletId}/payment/card";
```
Comment (mingazhev) [inline Api/Services/Payments/NGenius/NGeniusClient.cs:53]: Method is too large, need to split to smaller one for more readability

File: Api/Models/Payments/NGenius/BillingAddress.cs (hunk 1 above)
Comment (DFSko) [reply Api/Models/Payments/NGenius/BillingAddress.cs]: Fixed

File: Api/Services/Payments/NGenius/NGeniusClient.cs (hunk 2 above)
Comment (DFSko) [reply Api/Services/Payments/NGenius/NGeniusClient.cs:29]: Fixed

File: Api/Services/Payments/NGenius/NGeniusClient.cs (hunk 3 above)
Comment (DFSko) [reply Api/Services/Payments/NGenius/NGeniusClient.cs]: Fixed

**Review by mingazhev — APPROVED** [review]

**Pull request merged**